pyocd/target/builtin/target_CY8C6xx5.py

```diff
@@ -0,0 +1,77 @@
+# pyOCD debugger
+# Copyright (c) 2006-2013 Arm Limited
+# SPDX-License-Identifier: Apache-2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import logging
+from time import sleep
+
+from .flash_algo_CY8C6xx5 import flash_algo as flash_algo_main
+from .flash_algo_CY8C6xxA_WFLASH import flash_algo as flash_algo_work
+from .flash_algo_CY8C6xxA_SFLASH import flash_algo as flash_algo_sflash
+from .flash_algo_CY8C6xxA_SMIF_S25FL512S import flash_algo as flash_algo_smif
+from .target_CY8C6xxA import CY8C6xxA
+from ...core import exceptions
+from ...core.coresight_target import CoreSightTarget
+from ...core.memory_map import (FlashRegion, RamRegion, RomRegion, MemoryMap)
+from ...core.target import Target
+from ...coresight.cortex_m import CortexM
+from ...utility.timeout import Timeout
+
+LOG = logging.getLogger(__name__)
+
+ERASE_ALL_WEIGHT = 0.5 # Time it takes to perform a chip erase
+ERASE_SECTOR_WEIGHT = 0.05 # Time it takes to erase a page
+PROGRAM_PAGE_WEIGHT = 0.07 # Time it takes to program a page (Not including data transfer time)
+
+class CY8C6xx5(CY8C6xxA):
```

Comment:
It might be best to create a PSoC6 family class under `pyocd.target.family` since there are now a lot of target classes with common superclasses.